File: app/src/main/java/org/jaaksi/pickerview/demo/model/City.java
package org.jaaksi.pickerview.demo.model;

import java.util.List;
import org.jaaksi.pickerview.dataset.OptionDataSet;

/**
 * Created by fuchaoyang on 2018/2/11.<br/>
 * description：
 */

public class City implements OptionDataSet {

  public int id;
  public String name;
  public List<County> counties;

  @Override public CharSequence getCharSequence() {
    return name;
  }

  @Override public List<County> getSubs() {
    return counties;
  }

  @Override public String getValue() {
    return String.valueOf(id);
  }
}

File: app/src/main/java/org/jaaksi/pickerview/demo/model/City.kt
package org.jaaksi.pickerview.demo.model

import androidx.annotation.Keep
import org.jaaksi.pickerview.dataset.OptionDataSet

/**
 * Created by fuchaoyang on 2018/2/11.<br></br>
 * description：市
 */
@Keep
data class City(val name: String, val area: List<District>) : OptionDataSet {

    override fun getSubs() = area

    override fun getCharSequence() = name

    override fun getValue() = name
}